Console becomes unresponsive after several hours idle.

### Issue description
On both PocketMine-MP_1.6.1dev-73_794ff643_API-2.1.0 and PocketMine-MP_1.6.1dev-76_8214e50b_API-2.1.0, (Windows 10 Pro) after two or more hours of operation, unattended with no players in the game, upon returning to the console there is no response to commands.  The game is still functioning, players may login and the world responds normally.  However, at the console (the server terminal window), commands such as "list" or "version" are ignored.  Characters typed appear on the screen, but without effect.  Ctrl-C is recognized, terminating the program.

### Steps to reproduce the issue
<!--- help us find the problem by adding steps to reproduce the issue -->
1. start.cmd
2. verify operation of console (terminal window) by issuing "version" command.
3. verify game operation by briefly logging in from Windows Phone 8.1 device.
4. exit device from game.
5. leave system idle for two hours.
6. attempt "version" command again.
7. verify game operation by logging in from gaming device.

### OS and versions
[23:37:31] [Server thread/INFO]: This server is running PocketMine-MP 1.6.1dev-#76 「Unleashed」 implementing API version 2.1.0 for Minecraft: PE v0.16.0.5 alpha (protocol version 91)

* Title Bar:  PocketMine-MP 1.6.1dev-#76 | Online 0/20 | Memory 16.11/0/16.11/20 MB @ 6 threads | U 0 D 0 kB/s | TPS 20 | Load 0.28%

* PocketMine-MP:   PocketMine-MP_1.6.1dev-76_8214e50b_API-2.1.0

* PHP: PHP7.0.13

* Server OS:  Windows 10 Pro

* Game version: PE


### Crashdump, backtrace or other files
<!--- please use gist or anything else and add links here -->
* ...

this is a bug with stdin. I do encounter this bug with this script:
```php
	public function __construct(){
		global $stdin;
		$stdin = fopen("php://stdin", "r");
	}

	public function run(){
		while(!$this->doExit){
  			$foundInput = false;
  			while(!$foundInput){
  				global $stdin;
  				sleep(1);
				$input = trim(fgets($stdin));
				if($input != NULL || $input != ""){
					$foundInput = true;
				}else{
					echo("[ERROR] Stdin corrupted!".PHP_EOL);
					$stdin = fopen("php://stdin", "r");
					stream_set_blocking($stdin, 0);
				}
  			}
  			$this->processCMD($this->processRawInput($input));
		}
	}
```

--- a/src/pocketmine/command/CommandReader.php
+++ b/src/pocketmine/command/CommandReader.php
@@ -22,13 +22,13 @@
 namespace pocketmine\command;
 
 use pocketmine\Thread;
-use pocketmine\utils\Utils;
 
 class CommandReader extends Thread{
 	private $readline;
 	/** @var \Threaded */
 	protected $buffer;
 	private $shutdown = false;
+	private $streamBlocking = false;
 
 	public function __construct(){
 		$this->buffer = new \Threaded;
@@ -44,7 +44,7 @@ class CommandReader extends Thread{
 	private function initStdin(){
 		global $stdin;
 		$stdin = fopen("php://stdin", "r");
-		stream_set_blocking($stdin, 0);
+		$this->streamBlocking = (stream_set_blocking($stdin, 0) === false);
 	}
 
 	private function readLine(){
@@ -54,14 +54,15 @@ class CommandReader extends Thread{
 			if(!is_resource($stdin)){
 				$this->initStdin();
 			}
+
+			$line = fgets($stdin);
 			
-			$fgets = fgets($stdin);
-			
-			if(Utils::getOS() == "win" && $fgets === false){ //windows sucks
+			if($line === false and $this->streamBlocking === true){ //windows sucks
 				$this->initStdin();
+				$line = fgets($stdin);
 			}
 			
-			return trim($fgets);
+			return trim($line);
 		}else{
 			$line = trim(readline("> "));
 			if($line != ""){
@@ -90,7 +91,6 @@ class CommandReader extends Thread{
 			$this->initStdin();
 		}
 
-		$this->getClassLoader()->register(true);
 		$lastLine = microtime(true);
 		while(!$this->shutdown){
 			if(($line = $this->readLine()) !== ""){
@@ -103,8 +103,9 @@ class CommandReader extends Thread{
 
 			$lastLine = microtime(true);
 		}
+
 		global $stdin;
-		if(!$this->readline && is_resource($stdin)){
+		if(!$this->readline){
 			fclose($stdin);
 		}
 	}

--- a/src/pocketmine/command/CommandReader.php
+++ b/src/pocketmine/command/CommandReader.php
@@ -96,7 +96,9 @@ class CommandReader extends Thread{
 			$lastLine = microtime(true);
 		}
 		global $stdin;
-		fclose($stdin);
+		if(is_resource($stdin)){
+			fclose($stdin);
+		}
 	}
 
 	public function getThreadName(){

--- a/src/pocketmine/command/CommandReader.php
+++ b/src/pocketmine/command/CommandReader.php
@@ -22,6 +22,7 @@
 namespace pocketmine\command;
 
 use pocketmine\Thread;
+use pocketmine\utils\Utils;
 
 class CommandReader extends Thread{
 	private $readline;
@@ -51,10 +52,16 @@ class CommandReader extends Thread{
 			global $stdin;
 
 			if(!is_resource($stdin)){
-				$this->initStdin(); 
+				$this->initStdin();
 			}
-
-			return trim(fgets($stdin));
+			
+			$fgets = fgets($stdin);
+			
+			if(Utils::getOS() == "win" && $fgets === false){ //windows sucks
+				$this->initStdin();
+			}
+			
+			return trim($fgets);
 		}else{
 			$line = trim(readline("> "));
 			if($line != ""){
@@ -83,6 +90,7 @@ class CommandReader extends Thread{
 			$this->initStdin();
 		}
 
+		$this->getClassLoader()->register(true);
 		$lastLine = microtime(true);
 		while(!$this->shutdown){
 			if(($line = $this->readLine()) !== ""){
@@ -96,7 +104,7 @@ class CommandReader extends Thread{
 			$lastLine = microtime(true);
 		}
 		global $stdin;
-		if(is_resource($stdin)){
+		if(!$this->readline && is_resource($stdin)){
 			fclose($stdin);
 		}
 	}

--- a/src/pocketmine/command/CommandReader.php
+++ b/src/pocketmine/command/CommandReader.php
@@ -104,8 +104,8 @@ class CommandReader extends Thread{
 			$lastLine = microtime(true);
 		}
 
-		global $stdin;
 		if(!$this->readline){
+			global $stdin;
 			fclose($stdin);
 		}
 	}

--- a/src/pocketmine/command/CommandReader.php
+++ b/src/pocketmine/command/CommandReader.php
@@ -40,12 +40,18 @@ class CommandReader extends Thread{
 		$this->shutdown = true;
 	}
 
+	private function initStdin(){
+		global $stdin;
+		$stdin = fopen("php://stdin", "r");
+		stream_set_blocking($stdin, 0);
+	}
+
 	private function readLine(){
 		if(!$this->readline){
 			global $stdin;
 
 			if(!is_resource($stdin)){
-				return "";
+				$this->initStdin(); 
 			}
 
 			return trim(fgets($stdin));
@@ -74,9 +80,7 @@ class CommandReader extends Thread{
 
 	public function run(){
 		if(!$this->readline){
-			global $stdin;
-			$stdin = fopen("php://stdin", "r");
-			stream_set_blocking($stdin, 0);
+			$this->initStdin();
 		}
 
 		$lastLine = microtime(true);
@@ -91,6 +95,8 @@ class CommandReader extends Thread{
 
 			$lastLine = microtime(true);
 		}
+		global $stdin;
+		fclose($stdin);
 	}
 
 	public function getThreadName(){

